samples/captioning/multi-track-captions.html

```diff
@@ -19,22 +19,76 @@
             controlbar,
             player;
 
-        function startVideo() {
-            var url = FRAGMENTED_CAPTION_URL,
-                TTMLRenderingDiv;
-
+        function init() {
+            var TTMLRenderingDiv = document.querySelector("#ttml-rendering-div");
             videoElement = document.querySelector(".videoContainer video");
-            TTMLRenderingDiv = document.querySelector("#ttml-rendering-div");
-
             player = dashjs.MediaPlayer().create();
-            player.initialize(videoElement, url, true);
-            player.setTextDefaultEnabled(true);
-            player.setInitialMediaSettingsFor('fragmentedText', { 
-                lang: 'swe'
-            });
+            player.initialize(videoElement);
             player.attachTTMLRenderingDiv(TTMLRenderingDiv);
+            player.on('currentTrackChanged', function(e) { 
+                if (e.newMediaInfo.type === 'fragmentedText') {
+                    setTimeout(showDomStorage, 0);
+                }
+            });
             controlbar = new ControlBar(player); // Checkout ControlBar.js for more info on how to target/add text tracks to UI
             controlbar.initialize();
+            var checkbox = document.getElementById("textDefaultEnabled");
+            checkbox.indeterminate = true;
+            document.getElementById("lastMediaSettingsCachingInfoEnabled").checked = player.getSettings().streaming.lastMediaSettingsCachingInfo.enabled;
+            showDomStorage();
+        }
+
+        function showDomStorage() {
+            document.getElementById("domStorage").innerHTML = localStorage.getItem('dashjs_fragmentedText_settings');
+        }
+
+        function play() {
+            var url = FRAGMENTED_CAPTION_URL;
+            player.attachSource(url);
+            setTextDefaultEnabled();
+            setLang();
+            player.play();
+            showDomStorage();
+        }
+
+        function setTextDefaultEnabled() {
+            var checkbox = document.getElementById("textDefaultEnabled");
+            if (!checkbox.indeterminate) {
+                document.getElementById("textDefaultEnabledValue").innerHTML = checkbox.checked;
+                player.setTextDefaultEnabled(checkbox.checked);
+            } else {
+                document.getElementById("textDefaultEnabledValue").innerHTML = 'undefined';
```

Comment:
Undefined can be irritating here. I suggest we set the default to true, and also do not toggle the label for this checkbox. Result:
Either [x] Enabled or [] Enabled

Reply:
done